packages/react-sdk/src/core/components/ParticipantView/DefaultParticipantViewUI.tsx

```diff
@@ -0,0 +1,165 @@
+   */
+  showMenuButton?: boolean;
+} & Pick<ParticipantViewProps, 'participant'>;
+
+const ToggleButton = forwardRef<HTMLButtonElement, ToggleMenuButtonProps>(
+  (props, ref) => {
+    return <IconButton enabled={props.menuShown} icon="ellipsis" ref={ref} />;
+  },
+);
+
+export const DefaultParticipantViewUI = ({
+  participant,
+  indicatorsVisible = true,
+  menuPlacement = 'bottom-end',
+  showMenuButton = true,
+}: DefaultParticipantViewUIProps) => {
+  const call = useCall()!;
+  const { reaction, sessionId } = participant;
+
+  return (
+    <>
+      {showMenuButton && (
+        <MenuToggle
+          strategy="fixed"
+          placement={menuPlacement}
+          ToggleButton={ToggleButton}
+        >
+          <ParticipantActionsContextMenu participant={participant} />
+        </MenuToggle>
+      )}
+      {reaction && (
+        <Reaction reaction={reaction} sessionId={sessionId} call={call} />
+      )}
+      <ParticipantDetails
+        participant={participant}
+        indicatorsVisible={indicatorsVisible}
+      />
+    </>
+  );
+};
+
+export const ParticipantDetails = ({
+  participant,
+  indicatorsVisible = true,
+}: Pick<
+  DefaultParticipantViewUIProps,
+  'participant' | 'indicatorsVisible'
+>) => {
+  const {
+    isDominantSpeaker,
+    isLoggedInUser,
+    connectionQuality,
+    publishedTracks,
+    pinnedAt,
+    sessionId,
+    name,
+    userId,
+    videoStream,
+  } = participant;
+  const call = useCall()!;
+
+  const connectionQualityAsString =
+    !!connectionQuality &&
+    String(SfuModels.ConnectionQuality[connectionQuality]).toLowerCase();
+
+  const hasAudio = publishedTracks.includes(SfuModels.TrackType.AUDIO);
+  const hasVideo = publishedTracks.includes(SfuModels.TrackType.VIDEO);
+  const isPinned = !!pinnedAt;
+
+  const isDebugMode = useIsDebugMode();
+
+  return (
+    <div className="str-video__participant-details">
+      <span className="str-video__participant-details__name">
+        {name || userId}
+        {indicatorsVisible && isDominantSpeaker && (
+          <span
+            className="str-video__participant-details__name--dominant_speaker"
+            title="Dominant speaker"
+          />
+        )}
+        {indicatorsVisible && (
+          <Notification
+            isVisible={
+              isLoggedInUser &&
+              connectionQuality === SfuModels.ConnectionQuality.POOR
+            }
+            message="Poor connection quality. Please check your internet connection."
+          >
+            {connectionQualityAsString && (
+              <span
+                className={clsx(
+                  'str-video__participant-details__connection-quality',
+                  `str-video__participant-details__connection-quality--${connectionQualityAsString}`,
+                )}
+                title={connectionQualityAsString}
+              />
+            )}
+          </Notification>
+        )}
+        {indicatorsVisible && !hasAudio && (
```

Comment:
@arnautov-anton I am thinking, whether this should be handled via CSS and class name. Maybe somebody will want to keep the indicator there, just change the icon.

Reply:
I'm going to keep it as is for now as this is UI component which we should encourage users to build themselves. This PR aimed at this separation from the `ParticipantView(Box)`.